processing/src/main/java/org/apache/druid/segment/join/PossiblyNullColumnValueSelector.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.segment.join;
+
+import org.apache.druid.common.config.NullHandling;
+import org.apache.druid.query.monomorphicprocessing.RuntimeShapeInspector;
+import org.apache.druid.segment.ColumnValueSelector;
+
+import javax.annotation.Nullable;
+import java.util.function.BooleanSupplier;
+
+public class PossiblyNullColumnValueSelector<T> implements ColumnValueSelector<T>
+{
+  private final ColumnValueSelector<T> baseSelector;
+  private final BooleanSupplier beNull;
+  @Nullable
+  private final T nullValue;
+
+  PossiblyNullColumnValueSelector(final ColumnValueSelector<T> baseSelector, final BooleanSupplier beNull)
+  {
+    this.baseSelector = baseSelector;
+    this.beNull = beNull;
+    this.nullValue = NullHandling.defaultValueForClass(baseSelector.classOfObject());
+  }
+
+  @Override
+  public double getDouble()
+  {
+    return beNull.getAsBoolean() ? 0L : baseSelector.getDouble();
```

Comment:
should we be returning the `defaultValueForNull` instead of 0?

Reply:
Hmm, that's a boxed type, so probably not. But there's some `ZERO_*` constants that would be appropriate, so I'll use those.